config: allow setting stdout/err for exec

MithicSpirit opened the pull request:
Introduces the `exec_output` command, which sets the file to use for
stdout and stderr in the `exec` and `exec_always` commands.

This change is currently undocumented as I am still polling interest (it was
originally just for personal use, but a friend encouraged me to submit it
upstream). If it seems like it might be merged, I can write up some
documentation.

kennylevinsen commented:
What does this do that `exec "some_command >OUT 2>ERR"` wouldn't already handle?

MithicSpirit commented:
> What does this do that `exec "some_command >OUT 2>ERR"` wouldn't already handle?

Nothing, this is just for convenience so you don't have to specify that for every command.

llyyr commented:
> Nothing, this is just for convenience so you don't have to specify that for every command.

I think you probably _want_ to specify it for each command, otherwise mixing up stdout/stderr output from multiple different commands into a single file doesn't seem very useful.

MithicSpirit commented:
I mainly don't want Sway's logs (that are going to be piped into systemd-cat whenever I get around to it) to be cluttered with the logs from everything else. I can still manually override the output for specific commands if I want to, but that's very rare.

This does feel very niche, so I don't have high hopes for it being merged.

emersion commented:
You can achieve the same thing with the following:

```
set $exec exec >>sway-exec.log 2>&1

bindsym $mod+Return $exec $term
```

MithicSpirit commented:
Yeah, that's also an option. I think my command is more convenient though.

MithicSpirit committed:
config: allow setting stdout/err for exec

Introduces the exec_output command, which sets the file to use for
stdout and stderr in the exec and exec_always commands.
